### PR TITLE
chore: remove unreachable array case from hasSlot

### PR DESCRIPTION
**Pull request by mwcz**
## remove unreachable array case from hasSlot

The issue is described in #1567.  It's a fairly low priority issue, however it is blocking me from using PFE with [esbuild](https://esbuild.github.io/) in a new project.

When I looked into it further, I found that hasSlot was intended to accept slot name as a string, or an array of slot names (strings).  However, it's only being called with strings throughout the project.  Because of that, it seemed safe to simply remove the array case, rather than finding a way to make it reachable.  I updated the jsdoc and console messages to reflect that hasSlot now only accepts strings.

My plan is to `npm link` this branch locally, so I can get unblocked.  I figured I'd go ahead and submit it as a PR as well, but I'm totally fine with another solution if the array case is really needed.

### Related issues

- (#1567) [fix] hasSlot's array case is unreachable

### What has changed and why

- hasSlot's behavior has not changed
- the unreachable 'array' case of hasSlot is removed
- jsdoc comments and console logs related to hasSlot's arguments are updated to remove references to array arguments

### Testing instructions

Aside from automated tests, no testing is required, as no behavior has changed.

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).

Note: because this isn't changing any behavior, I'm choosing to flag this as a chore, not a fix, so there's no changelog entry.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

**Comment by castastrophe** [comment]
I know we're not using it, but I'd hate to regress the functionality.  Instead of accepting an array, what if it was a comma-separated list?

**Comment by mwcz** [comment]
> I know we're not using it, but I'd hate to regress the functionality. Instead of accepting an array, what if it was a comma-separated list?

We can certainly make proper arrays work without resorting to a comma-separation, the hasSlot function would just need a different approach to test whether the input is an array, since typeof can't return `"array"`.  I just wanted to go for the minimum viable change: removing the code that was both A) unreachable, and B) unused.  There's no regress in functionality because it already wasn't possible to use the array case.

I did try to "activate" the array case before removing it.  `name.constructor === Array` is one way to check for arrayness, but it can't slot into the existing switch statement, because it's switching on `typeof`.  I'd have had to rewrite the switch as a series of if/elses, but I wasn't comfortable reorganizing the bulk of the function since I'm not that familiar with it, and I didn't see any tests for it.

Just as a side note, here's a way to make single-arg functions like hasSlot work with arrays without having to program support directly:  `["slot1", "slot2"].every(s => this.hasSlot(s))` which is a bit more verbose, but pretty handy.

**Comment by mwcz** [comment]
Come to think of it, the switch statement could switch on `name.constructor` instead of `typeof  name`.  The cases could be `Array`, `String`, etc.  That would be a very straightforward change that would enable the intended array support.

**Comment by mwcz** [comment]
:open_mouth: and somehow, now, esbuild doesn't mind `typeof foo === "array"` and bundles just fine. :facepalm:

File: elements/pfelement/src/pfelement.js [hunk 1]
```diff
@@ -240,11 +240,12 @@ class PFElement extends HTMLElement {
   /**
    * Returns a boolean statement of whether or not that slot exists in the light DOM.
    *
+   * @param {String} name The slot name.
```
Comment (castastrophe) [inline elements/pfelement/src/pfelement.js:243]:
```suggestion
   * @param {String, Array} name The slot name.
```

File: elements/pfelement/src/pfelement.js [hunk 2]
```diff
@@ -240,11 +240,12 @@ class PFElement extends HTMLElement {
...
    * @example this.hasSlot("header");
    */
   hasSlot(name) {
     if (!name) {
-      this.warn(`Please provide at least one slot name for which to search.`);
+      this.warn(`Please provide a slot name to search for.`);
```
Comment (castastrophe) [inline elements/pfelement/src/pfelement.js:248]:
```suggestion
      this.warn(`Please provide at least one slot name for which to search.`);
```

**Review by castastrophe — APPROVED** [review]

**Pull request merged**